src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -340,23 +345,23 @@ private void ProcessDownloadedBlock(ChainedHeaderBlock chainedHeaderBlock)
             this.logger.LogTrace("(-)");
         }
 
-        private async Task OnPartialValidationCompletedCallbackAsync(PartialValidationResult validationResult)
+        private async Task OnPartialValidationCompletedCallbackAsync(ValidationContext validationContext)
         {
-            this.logger.LogTrace("({0}:'{1}')", nameof(validationResult), validationResult);
+            this.logger.LogTrace("({0}:'{1}')", nameof(validationContext), validationContext);
 
-            if (validationResult.Succeeded)
+            if (validationContext.Error == null)
             {
-                await this.OnPartialValidationSucceededAsync(validationResult.ChainedHeaderBlock).ConfigureAwait(false);
+                await this.OnPartialValidationSucceededAsync(new ChainedHeaderBlock(validationContext.BlockToValidate, validationContext.ChainedHeaderToValidate)).ConfigureAwait(false);
```

Comment:
Why all these changes that just introduce useless allocations all over the code?

Reply:
fixed